### PR TITLE
Recursion's homework

### PR DESCRIPTION
**Pull request by ArtstiomEzepchick**
Please check my recursion's homework.

### CODE REVIEW

File: 19_04/recursion_exercises.js [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+//     if (list.value) {
+//         console.log(list.value);
+//     };
+
+//     if (list.next) {
+//         console.log(printList(list.next));
+//     };
+// };
+
+// console.log(printList(list));
```
Comment (smilejke) [inline 19_04/recursion_exercises.js:197]: no need in `console.log` here

File: 19_04/recursion_exercises.js [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+
+// Второе решение
+
+// function printList(list) {
+//     if (list.value) {
+//         console.log(list.value);
+//     };
+
+//     if (list.next) {
+//         console.log(printList(list.next));
```
Comment (smilejke) [inline 19_04/recursion_exercises.js:193]: no need in `console.log` here

File: 19_04/recursion_exercises.js [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+// Первое решение
+
+// function printList(list) {
+//     while(list) { // Привык к циклу for - из-за этого еле додумался до цикла while, т.к. с for не получалось.
+//         console.log(list.value);
+//         list = list.next;
+//     };
+// };
+
+// console.log(printList(list));
```
Comment (smilejke) [inline 19_04/recursion_exercises.js:183]: no need in `console.log` here

**Review by smilejke — CHANGES_REQUESTED** [review]

File: 19_04/recursion_exercises.js [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+// function sumTo(n) { /*... ваш код ... */ }
+
+// alert( sumTo(100) ); // 5050
+// P.S. Какой вариант решения самый быстрый? Самый медленный? Почему?
+
+// P.P.S. Можно ли при помощи рекурсии посчитать sumTo(100000)?
+
+// ПЕРВОЕ РЕШЕНИЕ
+
+// function sumTo(n) {
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: renamed

File: 19_04/recursion_exercises.js [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+//     let result = 0;
+
+//     if (n === 1) {
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: done

File: 19_04/recursion_exercises.js [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+
+// ВТОРОЕ РЕШЕНИЕ
+
+// function sumTo(n) {
+
+//     if (n === 1) {
+//         return n;
+//     };
+
+//     return n + sumTo(n-1);
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: changed

File: 19_04/recursion_exercises.js [hunk 7]
```diff
@@ -0,0 +1,197 @@
...
+// 3! = 3 * 2 * 1 = 6
+// 4! = 4 * 3 * 2 * 1 = 24
+// 5! = 5 * 4 * 3 * 2 * 1 = 120
+// Задача – написать функцию factorial(n), которая возвращает n!, используя рекурсию.
+
+// alert( factorial(5) ); // 120
+// P.S. Подсказка: n! можно записать как n * (n-1)! Например: 3! = 3*2! = 3*2*1! = 6
+
+// function factorial(n) {
+
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js:92]: changed

File: 19_04/recursion_exercises.js [hunk 8]
```diff
@@ -0,0 +1,197 @@
...
+// Напишите функцию fib(n) которая возвращает n-е число Фибоначчи.
+
+// Пример работы:
+
+// function fib(n) { /* ваш код */ }
+
+// alert(fib(3)); // 2
+// alert(fib(7)); // 13
+// alert(fib(77)); // 5527939700884757
+// P.S. Все запуски функций из примера выше должны работать быстро. Вызов fib(77) должен занимать не более доли секунды.
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js:115]: Something wrong with solutions in this article. Recursion example is the same as mine, but others don't work as expected.
I took a new variation from "learnjs" website that is much more fastest than recursion. And it has a large description to analyse what is going on.

File: 19_04/recursion_exercises.js [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+//     }
+//   }
+// };
+// Напишите функцию printList(list), которая выводит элементы списка по одному.
+
+// Сделайте два варианта решения: используя цикл и через рекурсию.
+
+// Как лучше: с рекурсией или без?
+
+// let list = {
```
[the remark this answers is not in the document]
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: changed

File: 19_04/recursion_exercises.js (hunk 1 above)
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: done

File: 19_04/recursion_exercises.js (hunk 2 above)
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js:203]: done

File: 19_04/recursion_exercises.js (hunk 3 above)
Comment (ArtstiomEzepchick) [reply 19_04/recursion_exercises.js]: done

**Review by smilejke — APPROVED** [review]

**Pull request merged**